test/integration_test.py

```diff
@@ -978,23 +981,21 @@ def _get_indexed_bundles(self,
         for hit in hits:
             source = one(hit['sources'])
             for bundle in hit.get('bundles', ()):
-                bundle_fqid = dict(
-                    # FIXME: Encapsulate source JSON representations
-                    #        https://github.com/databiosphere/azul/issues/3889
-                    source=dict(id=source['sourceId'], spec=source['sourceSpec']),
-                    uuid=bundle['bundleUuid'],
-                    version=bundle['bundleVersion']
-                )
+                bundle_fqid = SourcedBundleFQIDJSON(uuid=bundle['bundleUuid'],
+                                                    version=bundle['bundleVersion'],
+                                                    source=SourceJSON(id=source['sourceId'],
+                                                                      spec=source['sourceSpec']))
                 if config.is_anvil_enabled(catalog):
                     is_supplementary = only(set(chain.from_iterable(
                         always_iterable(file['is_supplementary'])
                         for file in hit['files']
                     )), default=False)
+                    assert isinstance(bundle_fqid, AnvilBundleFQIDJSON)
```

Comment:
I presume you did not test these changes in a deployment that indexes AnVIL snapshots. 

`TypedDict` does not support subtype checking:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/nadove/.pyenv/versions/3.9.16/lib/python3.9/typing.py", line 1984, in __subclasscheck__
    raise TypeError('TypedDict does not support instance and class checks')
TypeError: TypedDict does not support instance and class checks
```

I presume the intent was to verify that `bundle_fqid.keys()` matches the keys declared in `AnvilBundleFQIDJSON`. That can't possibly be the case here, because the `entity_type` key is missing from the constructor call on line #984, and is not added until line #994.

Reply:
Good catch! Yes, I did not test this on AnVIL as I should have. The intent is to silence a type warning in PyCharm, which it does. I assumed `isinstance(bundle_fqid, AnvilBundleFQIDJSON)` would be treated like isinstance(bundle_fqid, dict) at runtime. Let me look at alternatives.